lib/qbo_api/connection.rb

```diff
@@ -101,11 +102,11 @@ def add_connection_adapter(conn)
     end
 
     def add_exception_middleware(conn)
-      conn.use FaradayMiddleware::RaiseHttpException
+      conn.use QboApi::RaiseHttpException
     end
 
     def add_authorization_middleware(conn)
-      conn.request :oauth2, access_token, token_type: 'bearer'
+      conn.request :authorization, access_token, token_type: 'bearer'
```

Comment:
Worth a comment how this relates to Faraday Middleware.

Reply:
Sure thing

Reply:
since

```ruby
  spec.add_runtime_dependency 'faraday', '>= 1.10.0'
````

we should support both 1.x and 2.x


see https://lostisland.github.io/faraday/middleware/authentication re Update Faraday 1.x->2.x auth code

In this case, it seems there was a missing oauth2 middleware?

and the change [should have been](https://github.com/minimul/qbo_api/pull/120)

```diff
-      conn.request :oauth2, access_token, token_type: 'bearer'
+      conn.request :authorization, 'Bearer', access_token
```

if we are to support only Faraday 2.x, I think. (didn't  confirm if this  new format works with Faraday 1.x)

in our own app, going from Faraday 1.x to 2.x in `exchange_authorization_code_for_access_token`

we had to change

```diff
    url = "https://oauth.platform.intuit.com"
      token_endpoint = "/oauth2/v1/tokens/bearer"

      connection = Faraday.new(url: url) { |conn|
-        conn.request :basic_auth, client_id, client_secret
+        conn.request :authorization, :basic, client_id, client_secret
        conn.request :url_encoded # application/x-www-form-urlencoded
        conn.headers["Accept"] = "application/json"
        conn.use Faraday::Response::Json, parser_options: {symbolize_names: true}
        conn.use Faraday::Response::RaiseError
        conn.response :detailed_logger, ::Accounting::Auth.logger if QboApi.log
        conn.adapter Faraday.default_adapter
      }
     raw_response = connection.post { |req|
        req.body = {code: authorization_code, grant_type: :authorization_code, redirect_uri: oauth2_redirect_uri}
        req.url token_endpoint
      }
      data = raw_response.body
      # {
      #   "expires_in": 3600,
      #   "x_refresh_token_expires_in": 8706412,
      #   "refresh_token": "newnew...",
      #   "access_token": "new...",
      #   "token_type": "bearer"
      # }
      yield data if block_given?
      new_refresh_token = data.fetch(:refresh_token)
      self.refresh_token = new_refresh_token
      self.access_token = data.fetch(:access_token)
      data
    rescue Faraday::ClientError => e
      ExceptionNotifier.error(e, "[Quickbooks][#{qualified_name}] Authorization 2 token exchange failure",
       ....
```


and similarly in our `refresh_access_token` method